src/scripts/main.js

```diff
@@ -1,11 +1,29 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    if (element) {
+      element.addEventListener(
+        eventName,
+        (e) => {
+          resolve(
+            `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+          );
+        },
+        { once: true },
+      );
+    } else {
+      reject(new Error('Element not found'));
+    }
+  });
```

Comment:
Have you ever heard about guard clause? 
```suggestion
   if (!element) {
      return reject(new Error('Element not found'));
    }

    element.addEventListener(eventName, (e) => {
      resolve(
        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
      );
    }, { once: true });
```

We can check if the element exists before adding the event listener and rejects the promise if the element is not found.